### DISCUSSION TITLE
6.2 My Layer Previews Are White???

### DISCUSSION THREAD

**Discussion by andreasrf** [ua-gist604b-f24/syllabus#30] [General]
In another series of unfortunate events with Module 6, all my layer previews for the very last step of assignment 6-2 are white. I've tried redoing the assignment twice now with a brand new codespace, re-downloading my data, and connecting to my OSM directly with the terminal. I was able to load up the OSM just fine on my Codespace. All the layers are on Geoserver, only the previews are scuffed.

osm:osm only loads to a world map, which I'm not sure is correct.
> ![image](https://github.com/user-attachments/assets/7ef9ab6a-1e8c-483e-9ac9-998e2d0cf0fa)

> ![image](https://github.com/user-attachments/assets/11663144-a3e0-472b-841a-ac2e3ce2d7e3)

> ![image](https://github.com/user-attachments/assets/eee89693-a4ca-4cc3-8644-ac0d30576b6d)

**Comment by redmesa** [comment]
Hello, I'm also confused on this, but if I remove the layer and re-add it, the data does show up.
![image](https://github.com/user-attachments/assets/5abd1907-7ac4-4fa8-bb75-9fc543ad5172)

I verified that the data was copied over to the gostGis container.
![image](https://github.com/user-attachments/assets/38c6adf1-301f-47b7-a83c-dbc3cf6e52f2)

So I see the same thing you are seeing. 

Any hints or direction would be helpful

**Reply by kev-bone** [reply]
This is where I am at as well. Which is technically an improvement, since last night the layer previews wouldn't even open. It kept trying to download an unknown file each time.

**Reply by aldunn4** [reply]
Mine is doing the same. If I try and use OpenLayers it just tries to download a file called wms instead of opening the map in a new tab like it did before.

**Reply by kev-bone** [reply]
Yeah exactly, maybe wait a bit and try again.

**Reply by aaryno** [reply]
I will share a video walkthrough of this. There is an error that we need to solve. If you look at the "wms" file it will expose a more specific error relating to missing data from one of the layers. It will need to be opened in a text editor

**Reply by redmesa** [reply]
@aldunn4 I had this issue, but it was because my Plug-in didn't download. If you open the file in a text editor, it will give you more information on why it's failing.

**Comment by kev-bone** [comment]
Of course now that I say it didn't work, maybe it did? Individual layer previews are still blank/white. But on the osm:osm preview, try zooming in to Hawaii @andreasrf and @redmesa. Everywhere else is just beige polygons, but Hawaii does have the OSM style!

![image](https://github.com/user-attachments/assets/31b9477e-57c1-473b-8399-781445a0f39e)

**Reply by aaryno** [reply]
The WMS error relates to a scale-dependent layer that is broken. If you look more closely at the "wms" file in a text editor it will tell us what the error is. That will give us a clue as to how to proceed.

**Reply by redmesa** [reply]
Hello,

I believe that part of the assignment you know is working for Hawaii; now get it working for the requested area.

**Reply by aaryno** [reply]
@kev-bone - looks good. Watch the 6-2 lecture for a discussion on how that works and why. And for individual layer previews they are initially zoomed way out to the extent of the world (-180, -90, 90, 180) but the data only exists for Hawaii. Remember that this geoserver data directory was pre-initialized for us and it had no idea what the layer extents were. You can recalculate the layer extents by Editing the layer and re-calculating the BBOXes. Don't forget to save!
